imod/mf6/rch.py

```diff
@@ -152,3 +161,68 @@ def _validate(self, schemata, **kwargs):
 
     def get_regrid_methods(self) -> Optional[dict[str, Tuple[RegridderType, str]]]:
         return self._regrid_method
+
+    @classmethod
+    def from_imod5_data(
+        cls,
+        imod5_data: dict[str, dict[str, GridDataArray]],
+        target_grid: GridDataArray,
+        regridder_types: Optional[dict[str, tuple[RegridderType, str]]] = None,
+    ) -> "Recharge":
+        """
+        Construct an rch-package from iMOD5 data, loaded with the
+        :func:`imod.formats.prj.open_projectfile_data` function.
+
+        .. note::
+
+            The method expects the iMOD5 model to be fully 3D, not quasi-3D.
+
+        Parameters
+        ----------
+        imod5_data: dict
+            Dictionary with iMOD5 data. This can be constructed from the
+            :func:`imod.formats.prj.open_projectfile_data` method.
+        target_grid: GridDataArray
+            The grid that should be used for the new package. Does not
+            need to be identical to one of the input grids.
+        regridder_types: dict, optional
+            Optional dictionary with regridder types for a specific variable.
+            Use this to override default regridding methods.
+
+        Returns
+        -------
+        Modflow 6 rch package.
+
+        """
+
+        data = {
+            "rate": imod5_data["rch"]["rate"],
+        }
+        new_package_data = {}
+
+        # first regrid the inputs to the target grid.
+        regridder_settings = deepcopy(cls._regrid_method)
+        if regridder_types is not None:
+            regridder_settings.update(regridder_types)
+
+        regrid_context = RegridderWeightsCache()
+
+        new_package_data = _regrid_package_data(
+            data, target_grid, regridder_settings, regrid_context, {}
+        )
+
+        # if rate has only layer 0, then it is planar.
+        if is_planar_grid(data["rate"]):
+            # create an array indicating in which cells rch is active
+            is_rch_cell = allocate_rch_cells(
+                ALLOCATION_OPTION.at_first_active,
+                target_grid,
```

Comment:
The ``active`` grid (``idomain == 1``) should be inserted here. This comes from the regridded discretization package. So you therefore require an extra argument to the ``from_imod5_data`` method, namely the regridded discretization package.

For an example of the API of the ``allocate_<pkg>_cells``, see: https://deltares.github.io/imod-python/user-guide/09-topsystem.html#allocate-river-cells

Reply:
done